lib/senkyoshi/models/heirarchy.rb

```diff
@@ -0,0 +1,81 @@
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  class Heirarchy
+    def self.item_iterator(item, course_toc, discussion_boards)
+      if item.search("item").count.zero?
+        toc_item = setup_item(item, item.parent, course_toc)
+        toc_item[:indent] = 0
+        set_discussion_boards(discussion_boards, toc_item)
+      else
+        item.search("item").flat_map do |internal_item|
+          toc_item = setup_item(internal_item, item, course_toc)
+          toc_item[:indent] = get_indent(internal_item)
+          toc_item = set_discussion_boards(discussion_boards, toc_item)
+          toc_item
+        end
+      end
+    end
+
+    def self.get_indent(item, indent = -2)
+      return indent if item.parent.name == "organization"
+      indent += 1
+      get_indent(item.parent, indent)
+    end
+
+    def self.set_discussion_boards(discussion_boards, toc_item)
+      if toc_item[:internal_handle] == "discussion_board_entry"
+        resource = discussion_boards.select do |db|
+          title_attribute = db.attributes["title"] || db.attributes["bb:title"]
+          title_attribute.value == toc_item[:title]
+        end
+        if resource.count == 1
```

Comment:
Then it does nothing because you can't guarantee that it will connect the right discussion to the right location.